dev/src/write-batch.ts

```diff
@@ -577,7 +586,54 @@ export class WriteBatch implements firestore.WriteBatch {
       api.BatchWriteResponse
     >('batchWrite', request, tag, retryCodes);
 
-    return (response.writeResults || []).map((result, i) => {
+    let retryCount = 0;
+    while (retryCount < MAX_BATCH_WRITE_RETRY_ATTEMPTS) {
+      // Find the indexes of all writes that failed with ABORTED.
+      const abortedIndexes = response.status.reduce(
+        (arr: number[], status, i) => {
+          if (status.code === Status.ABORTED) {
+            arr.push(i);
```

Comment:
Ideally, we would combine this with loop below. This might work:

```
const writesToRetry  = response.status.reduce((writesToRetry, status, i) => {
    if (shouldRetry(status)) {
       writesToRetry.push(writes[i]);
    }
   return writesToRetry
}, []);
```